### PR TITLE
Add cassandra feature to Mariner

### PR DESCRIPTION
**Pull request by chalamalasetty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add cassandra feature to Mariner. 

###### Change Log  

- Change
   Add cassandra feature to Mariner. 

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local build.

### CODE REVIEW

File: SPECS-EXTENDED/cassandra/Readme [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+1. Boot up Mariner Core 2.0
+2. Refer to cassandra spec and install all build required dependenices.
+3. Download cassandra sources and extract under /usr/src
+4. cd /usr/src/apache-cassandra-<version>-src
+5. Run "export JAVA_HOME=/usr/lib/jvm/msopenjdk-11"
+6. Run export ANT_OPTS="-Xmx1024m -XX:MaxPermSize=512m"
+7. Run "sudo ant clean ant -v clean jar -Drelease=true -Duse.jdk11=true"
+8. Once build completed successfully take build cache as below.
+9. cd ~/.m2
+9. tar -czvf cassandra-build-cache-<version>.tar.gz repository
```
Comment (anphel31) [inline SPECS-EXTENDED/cassandra/Readme:14]: Suggest to use tar commandline arguments used in other specs to make this reproducible. See - https://github.com/microsoft/CBL-Mariner/blob/9f9217c5f9f55621dec5a188cba780c5c8972a1b/SPECS/azure-iotedge/azure-iotedge.spec#L14

**Review by anphel31 — APPROVED** [review]

File: SPECS-EXTENDED/cassandra/Readme (hunk 1 above)
Comment (chalamalasetty) [reply SPECS-EXTENDED/cassandra/Readme]: updated.

**Review by neha170 — APPROVED** [review]

File: SPECS-EXTENDED/cassandra/cassandra.spec [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+Version:        4.0.3
+Release:        1%{?dist}
+URL:            http://cassandra.apache.org/
+License:        Apache License, Version 2.0
+Group:          Applications/System
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Source0:        http://archive.apache.org/dist/cassandra/%{version}/apache-%{name}-%{version}-src.tar.gz
+Source1:        cassandra.service
+Source2:        cassandra-build-cache-4.0.3.tar.gz
```
Comment (PawelWMS) [inline SPECS-EXTENDED/cassandra/cassandra.spec:15]: In addition to the `%prep` section, could you also mention that re-build instructions are in the readme file?

File: SPECS-EXTENDED/cassandra/cassandra.spec [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+Summary:        Cassandra is a highly scalable, eventually consistent, distributed, structured key-value store
+Name:           cassandra
+Version:        4.0.3
+Release:        1%{?dist}
+URL:            http://cassandra.apache.org/
+License:        Apache License, Version 2.0
+Group:          Applications/System
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Source0:        http://archive.apache.org/dist/cassandra/%{version}/apache-%{name}-%{version}-src.tar.gz
```
Comment (PawelWMS) [inline SPECS-EXTENDED/cassandra/cassandra.spec:13]:
```suggestion
Source0:        https://archive.apache.org/dist/cassandra/%{version}/apache-%{name}-%{version}-src.tar.gz
```

File: SPECS-EXTENDED/cassandra/cassandra.spec [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+%build
+# refer to Readme file for 
+# cassandra-build-cache
+cp %{SOURCE2} .
+tar -xvf  %{SOURCE2}
+mkdir -p ~/.m2
+mv repository ~/.m2/
+
+export JAVA_HOME=$(echo /usr/lib/jvm/msopenjdk-11)
```
Comment (PawelWMS) [inline SPECS-EXTENDED/cassandra/cassandra.spec:62]: Please use `%{java_home}` instead.

```suggestion
export JAVA_HOME="%{java_home}"
```

File: SPECS-EXTENDED/cassandra/cassandra.spec [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+%{_sbindir}/ldconfig
+%systemd_postun_with_restart cassandra.service
+if [ $1 -eq 0 ] ; then
+    /usr/sbin/userdel cassandra
+    /usr/sbin/groupdel cassandra
+fi
+
+%files
+%defattr(-,root,root)
+%doc README.asc CHANGES.txt NEWS.txt conf/cqlshrc.sample LICENSE.txt NOTICE.txt
```
Comment (PawelWMS) [inline SPECS-EXTENDED/cassandra/cassandra.spec:125]: Please declare license-related files with the `%license` macro:

```suggestion
%license LICENSE.txt
%doc README.asc CHANGES.txt NEWS.txt conf/cqlshrc.sample NOTICE.txt
```

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Few short comments, I'm mostly asking to replace the hard-coded Java path.

File: SPECS-EXTENDED/cassandra/cassandra.spec (hunk 3 above)
Comment (chalamalasetty) [reply SPECS-EXTENDED/cassandra/cassandra.spec]: updated.

File: SPECS-EXTENDED/cassandra/cassandra.spec (hunk 4 above)
Comment (chalamalasetty) [reply SPECS-EXTENDED/cassandra/cassandra.spec]: updated

File: SPECS-EXTENDED/cassandra/cassandra.spec (hunk 5 above)
Comment (chalamalasetty) [reply SPECS-EXTENDED/cassandra/cassandra.spec]: added %license LICENSE.txt

File: SPECS-EXTENDED/cassandra/cassandra.spec (hunk 2 above)
Comment (chalamalasetty) [reply SPECS-EXTENDED/cassandra/cassandra.spec:15]: Added comment in %prep to refer to Readme file for detailed instructions.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**